### PR TITLE
HPCC-31706 Fix Apple Clang warnings and errors on Apple silicon builds

### PR DESCRIPTION
**Pull request by timothyklemm**
- Add missing format attributes.
- Remove extraneous parentheses.
- Remove set but unused variables.
- Remove reference to obsolete STL base class.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp [hunk 1]
```diff
@@ -3454,7 +3454,9 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                 // NB: when preserveGroups, the lhs group will always be complete at same time, so this will traverse whole group
                 if (!joinGroup->hasFlag(GroupFlags::gf_head))
                     return; // intermediate rows are completing, but can't output any of those until head finishes, at which point head marker will shift to next if necessary (see below)
+                #ifdef _DEBUG
```
Comment (jakesmith) [inline thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp:3457]: trivial/style: we normally don't indent #if*'s like this

**Review by jakesmith — APPROVED** [review]
@timothyklemm - looks good.

File: thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp (hunk 1 above)
Comment (timothyklemm) [reply thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp]: Pushed a squashed update that only changed the indentation.

**Comment by timothyklemm** [comment]
@ghalliday this should be ready to merge.

**Pull request merged**